### ISSUE TITLE
Check if app can access a file

### ISSUE THREAD

**Issue by matthiasclasen** [alexlarsson/xdg-app-gtk#7]
When an app has partial access to the file system, say the xdg-documents folder, we should try detect when the selected file is in that folder, and not go through the documents portal in that case.

**Comment by alexlarsson** [comment]
Yeah, this requires us to pass through a lot more information about the app in the portals calls though...

**Comment by alexlarsson** [comment]
Hmm, its easy for the app to figure out what is readable (its in /run/user/1000/flatpak-info), but the portal can't really trust such info from the app, because it could say it can see everything and then get back the full pathname of the selected app, leaking some (minor) info about the host fs to the sandbox.

So, we need to get the portal an accurate copy of the sandboxed permissions somehow.
